src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -1118,6 +1118,14 @@ public void Trace(SocketAsyncContext context, string message, [CallerMemberName]
 
         private readonly object _registerLock = new object();
 
+        public static readonly SocketAsyncContext Closed;
+
+        static SocketAsyncContext()
+        {
+            Closed = new SocketAsyncContext(new SafeSocketHandle());
+            Closed.Close();
+        }
```

Comment:
Adding an explicit static cctor adds overhead to other static accesses on the type.  It'd be better to do this as:
```C#
internal static readonly SocketAsyncContext Closed = CreateClosedContext();

private static SocketAsyncContext CreateClosedContext()
{
    var closed = new SocketAsyncContext(new SafeSocketHandle());
    closed.Close();
    return closed;
}
```